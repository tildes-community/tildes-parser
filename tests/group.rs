use std::fs::read_to_string;

use {insta::assert_debug_snapshot, scraper::Html, tildes_parser::Group};

#[test]
fn test_group_parsing() {
  let html = read_to_string("tests/samples/group.html").unwrap();
  let html = Html::parse_document(&html);
  let group = Group::from_html(&html).unwrap();
  assert_debug_snapshot!(group);
}
