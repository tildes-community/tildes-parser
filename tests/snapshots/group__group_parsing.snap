---
source: tests/group.rs
expression: group
---
Group {
    description: Some(
        "Group description.",
    ),
    name: "~group",
    sub_groups: [
        "~example.sub",
    ],
    subscribers: 12345,
    wiki_links: [
        GroupWikiLink {
            name: "index",
            url: "https://example.org/~example/wiki/index",
        },
        GroupWikiLink {
            name: "Example Page",
            url: "https://example.org/~example/wiki/example_page",
        },
    ],
}
